gitpod-io/gitpod
# Allow setting customTimeoutAnnotation for headless workspace pods

This changes two things:

- In `server`: Don't set `customTimeoutAnnotation` when starting headless pods (currently ignored by ws-manager)
- In `ws-manager`: Honor `customTimeoutAnnotation` for headless pods (when it is set)

See also [Slack discussion](https://gitpod.slack.com/archives/C0204GGRKT4/p1621342252000100) (internal)

And also (optionally):

- In `ws-manager`: When a workspace times out, log both how long it took and the expected timeout (new)

### How to test

1. Start a regular workspace (for any repo)
2. The workspace pod should still have a `gitpod/customTimeout` label (e.g. `30m` if you're on the Open Source / Free plan)
3. Start a prebuild (e.g. using the `#prebuild/` manual prefix)
4. The prebuild pod should **not** have a `gitpod/customTimeout` label
5. If you manually set a `gitpod/customTimeout` label on the prebuild pod (e.g. `1m`), the prebuild should honor it instead of the default headless timeout (which is `60m`)
